go/iot_sub.go

```diff
@@ -0,0 +1,4 @@
+// Subscribe to messages sent on the channel with given eventName
```

Comment:
This snippet needs to use an MQTT client not an Ably client (see the [equivalent JavaScript snippet](https://github.com/ably/homepage-snippets/blob/main/javascript/iot-sub.js)), and please drop the comment